mhlib/app_test.go

```diff
@@ -0,0 +1,67 @@
+package mhlib
+
+import (
+	"testing"
+)
+
+
+func TestSelfRender(t *testing.T) {
+
+	// With datasources
+	templateString := `
+gomplate:
+  datasources:
+  - "http_obj=https://httpbin.org/get"
+  datasourceheaders: []
+values:
+  a: foo
+  b: '[[ .values.a ]]bar' # foobar
+  c: '[[ .values.b ]]baz' # foobarbaz
+  f: 'Func Test: [[ net.LookupIP "example.com" ]]'
+`
+	expected := `
+gomplate:
+  datasources:
+  - "http_obj=https://httpbin.org/get"
+  datasourceheaders: []
+values:
+  a: foo
+  b: 'foobar' # foobar
+  c: 'foobarbaz' # foobarbaz
+  f: 'Func Test: 10.114.236.103'
```

Comment:
I rushed this together to try and figure out what the function was doing, this address probably should be looked up, `net.dial`, instead of hard coded in expected.

```
import "net"
ips, err := net.LookupIP("google.com")
if err != nil {
  t.Log(err)
  t.Fail()
}
// use ips[0].string() instead pf 10.114.236.103
```

But if its worth adding tests, I can make them clean in a new PR.